src/commands/force/source/delete.ts

```diff
@@ -0,0 +1,204 @@
+    if (this.aborted) return;
+
+    // fire predeploy event for the delete
+    await this.lifecycle.emit('predeploy', this.componentSet.toArray());
+    this.isRest = await this.isRestDeploy();
+    this.ux.log(`*** Deleting with ${this.isRest ? 'REST' : 'SOAP'} API ***`);
+
+    const deploy = await this.componentSet.deploy({
+      usernameOrConnection: this.org.getUsername(),
+      apiOptions: {
+        rest: this.isRest,
+        checkOnly: this.getFlag<boolean>('checkonly', false),
+        testLevel: this.getFlag<TestLevel>('testlevel'),
+      },
+    });
+    this.updateDeployId(deploy.id);
+
+    if (!this.isJsonOutput()) {
+      const progressFormatter: ProgressFormatter = env.getBoolean('SFDX_USE_PROGRESS_BAR', true)
+        ? new DeployProgressBarFormatter(this.logger, this.ux)
+        : new DeployProgressStatusFormatter(this.logger, this.ux);
+      progressFormatter.progress(deploy);
+    }
+
+    this.deployResult = await deploy.pollStatus(500, this.getFlag<Duration>('wait').seconds);
+    await this.lifecycle.emit('postdeploy', this.deployResult);
+  }
+
+  /**
+   * Checks the response status to determine whether the delete was successful.
+   */
+  protected resolveSuccess(): void {
+    const status = getString(this.deployResult, 'response.status');
+    if (status !== RequestStatus.Succeeded && !this.aborted) {
+      this.setExitCode(1);
+    }
+  }
+
+  protected formatResult(): DeployCommandResult {
+    const formatterOptions = {
+      verbose: this.getFlag<boolean>('verbose', false),
+    };
+
+    this.deleteResultFormatter = new DeleteResultFormatter(this.logger, this.ux, formatterOptions, this.deployResult);
+
+    // Only display results to console when JSON flag is unset.
+    if (!this.isJsonOutput()) {
+      this.deleteResultFormatter.display();
+    }
+
+    return this.deleteResultFormatter.getJson();
+  }
+
+  private deleteFilesLocally(): void {
+    if (!this.getFlag('checkonly') && getString(this.deployResult, 'response.status') === 'Succeeded') {
+      this.sourceComponents.map((component) => {
+        // delete the content and/or the xml of the components
+        if (component.content) {
+          const stats = fs.lstatSync(component.content);
+          if (stats.isDirectory()) {
+            fs.rmdirSync(component.content, { recursive: true });
+          } else {
+            fs.unlinkSync(component.content);
+          }
+        }
+        // the xml could've been deleted as part of a bundle type above
+        if (component.xml && fs.existsSync(component.xml)) {
+          fs.unlinkSync(component.xml);
+        }
+      });
+    }
+  }
+
+  private async handlePrompt(): Promise<boolean> {
+    if (!this.getFlag('noprompt')) {
+      const paths = this.sourceComponents.map((component) => component.content + '\n' + component.xml).join('\n');
+      const promptMessage = messages.getMessage('prompt', [paths]);
+      const answer = (await prompt(promptMessage)) as string;
```

Comment:
I've always wondered why CLI commands use `prompt` instead of `confirm` (and then have to do the extra casing, etc).

source: https://github.com/oclif/cli-ux/blob/fd0ca55f2ae7096ea273592be35e071ee4ddcb0d/src/prompt.ts#L132

ex in use
https://github.com/forcedotcom/source-tracking/blob/e73b17609ae410be4043b423b8488968a7513afe/src/commands/force/source/tracking/reset.ts#L44

Reply:
Oh, it's probably just a carry-over from TB, I'll swap it for `confirm`